### PR TITLE
Bitbucketstatus working

### PR DESCRIPTION
**Pull request by cwadley**

### CODE REVIEW

File: plugins/inputs/bitbucket_status/bitbucket_status.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	"log"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"time"
+)
+
+type BitbucketStatus struct {
+	StatusMeas 			string 	`toml:"status_measurement"`
```
Comment (kepoling) [inline plugins/inputs/bitbucket_status/bitbucket_status.go]: Formatting is messed up here, I can't tell if its just being rendered weird or its actually like this. Was gofmt ran on this code?

Comment (cwadley) [reply plugins/inputs/bitbucket_status/bitbucket_status.go]: Thought I had run it, but I guess not. Updated.

File: plugins/inputs/bitbucket_status/bitbucket_status.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
+package bitbucket_status
+
+import (
+	"log"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"time"
```
Comment (kepoling) [inline plugins/inputs/bitbucket_status/bitbucket_status.go]: Per convention, can you please move this import to the top with "log" and "sync"?

Comment (cwadley) [reply plugins/inputs/bitbucket_status/bitbucket_status.go]: Whoops. Done.

File: plugins/inputs/bitbucket_status/bitbucket_status.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+func AccumulateComponents(bbStatus *BBStatus, measName string, acc telegraf.Accumulator) {
+	batchWriteTime := time.Now()
+	for _, component := range bbStatus.Components {
+		var statusInt int
+		if component.Status == "operational" {
+			statusInt = 0
+		} else if component.Status == "partial_outage" {
+			statusInt = 1
+		} else if component.Status == "major_outage" {
```
Comment (kepoling) [inline plugins/inputs/bitbucket_status/bitbucket_status.go]: If there are only 3 statuses I would arrange this to default to the operational case and get rid of the last `else if` in favor of just an `else`.

Comment (cwadley) [reply plugins/inputs/bitbucket_status/bitbucket_status.go]: Your wish is my command. Good thing you said something too, cause it turns out I missed a possible status, so I added that as well.

File: plugins/inputs/bitbucket_status/bitbucket_status_client.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+func (bsc *BitbucketStatusClient) GetStatus(requestUrl string) (*BBStatus, error) {
+	req, err := BuildRequest(requestUrl)
+	if err != nil {
+		log.Fatal("There was an error building the request for bitbucket status: ", err)
+		return nil, err
+	}
+
+	responseBody, err := MakeRequest(req)
+	if err != nil {
+		log.Fatal("There was an error making the request for bitbucket status: ", err)
```
Comment (kepoling) [inline plugins/inputs/bitbucket_status/bitbucket_status_client.go]: Do we want to totally kill the collector if something happens during the request?

File: plugins/inputs/bitbucket_status/bitbucket_status_client.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	status := ParseRequest(responseBody)
+	return status, nil
+}
+
+func ParseRequest(responseBody []byte) *BBStatus {
+	status := new(BBStatus)
+
+	err := json.Unmarshal(responseBody, status)
+	if err != nil {
+		log.Fatal("There was an error unmarshalling json for bitbucket status: ", err)
```
Comment (kepoling) [inline plugins/inputs/bitbucket_status/bitbucket_status_client.go]: Is this worth killing the collector?

File: plugins/inputs/bitbucket_status/bitbucket_status_client.go (hunk 4 above)
Comment (cwadley) [reply plugins/inputs/bitbucket_status/bitbucket_status_client.go]: Nope, don't want to kill telegraf.

File: plugins/inputs/bitbucket_status/bitbucket_status_client.go (hunk 5 above)
Comment (cwadley) [reply plugins/inputs/bitbucket_status/bitbucket_status_client.go]: Nope. Fixed.

**Pull request closed without merging**